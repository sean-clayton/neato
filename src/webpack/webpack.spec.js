/* eslint-env jest */

import webpackConfig from './webpack'

describe('webpack', () => {
  it('Creates a webpack config object', () => {
    const neatoConfigMock = {
      options: {
        port: 3000,
        pages: ['index']
      }
    }
    expect(webpackConfig(neatoConfigMock)).toMatchSnapshot()
  })

  it('Accepts a custom config', () => {
    const neatoConfigMock = {
      options: {
        port: 3000,
        pages: ['index'],
        webpack: {
          output: {
            publicPath: 'https://cdn.com/'
          }
        }
      }
    }
    expect(webpackConfig(neatoConfigMock)).toMatchSnapshot()
  })
})
