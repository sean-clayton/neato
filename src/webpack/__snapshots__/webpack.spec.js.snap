exports[`webpack Accepts a custom config 1`] = `
Object {
  "devtool": "source-map",
  "entry": Object {
    "index": "./index",
    "vendor": Array [],
  },
  "output": Object {
    "publicPath": "https://cdn.com/",
  },
  "plugins": Array [
    NoErrorsPlugin {},
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": true,
        "inject": true,
        "minify": false,
        "showErrors": true,
        "template": "index.html",
        "title": "Webpack App",
        "xhtml": false,
      },
    },
    CommonsChunkPlugin {
      "async": undefined,
      "chunkNames": Array [
        "shared",
        "vendor",
        "webpack",
      ],
      "filenameTemplate": undefined,
      "ident": "/home/sean/code/neato/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js1",
      "minChunks": undefined,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
  ],
  "resolve": Object {
    "modules": Array [],
  },
}
`;

exports[`webpack Creates a webpack config object 1`] = `
Object {
  "devtool": "source-map",
  "entry": Object {
    "index": "./index",
    "vendor": Array [],
  },
  "output": Object {
    "publicPath": "/",
  },
  "plugins": Array [
    NoErrorsPlugin {},
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": true,
        "inject": true,
        "minify": false,
        "showErrors": true,
        "template": "index.html",
        "title": "Webpack App",
        "xhtml": false,
      },
    },
    CommonsChunkPlugin {
      "async": undefined,
      "chunkNames": Array [
        "shared",
        "vendor",
        "webpack",
      ],
      "filenameTemplate": undefined,
      "ident": "/home/sean/code/neato/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js0",
      "minChunks": undefined,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
  ],
  "resolve": Object {
    "modules": Array [],
  },
}
`;
